### PR TITLE
fixed indicator position in code with tabs

### PR DESCRIPTION
**Pull request by dosisod**
PR for issue #8306.

### CODE REVIEW

**Comment by dosisod** [comment]
Now that the tests are actually failing for code related reasons, what is the suggested change for the 2 failing tests?

https://github.com/python/mypy/blob/5d8c0cd28f8c038e853731d2cd12578412945beb/test-data/unit/fine-grained-blockers.test#L27

The cursor is off by 1 character, though I would think that the new position of the cursor makes more sense. Also, where should I put new tests to check for consistency between tabs and spaces when pretty printing?

File: mypy/errors.py [hunk 1]
```diff
@@ -451,12 +451,17 @@ def format_messages(self, error_info: List[ErrorInfo],
                 # Add source code fragment and a location marker.
                 if severity == 'error' and source_lines and line > 0:
                     source_line = source_lines[line - 1]
+                    source_line_expanded = source_line.replace('\t', ' ' * 8)
```
Comment (ilevkivskyi) [inline mypy/errors.py]: Why not `source_line.expandtabs()`? (Also 8 is default for the latter.)

File: mypy/errors.py [hunk 2]
```diff
@@ -451,12 +451,17 @@ def format_messages(self, error_info: List[ErrorInfo],
...
                     if column < 0:
                         # Something went wrong, take first non-empty column.
                         column = len(source_line) - len(source_line.lstrip())
+
+                    # Shifts column after tab expansion
+                    column = len(source_line[:column].replace('\t', ' ' * 8))
```
Comment (ilevkivskyi) [inline mypy/errors.py]: Same question here.

**Comment by ilevkivskyi** [comment]
> The cursor is off by 1 character, though I would think that the new position of the cursor makes more sense.

This doesn't look important, you can just update the existing tests.

> Also, where should I put new tests to check for consistency between tabs and spaces when pretty printing?

I would put them at the end of `cmdline.test`.

**Comment by dosisod** [comment]
Did not know that `expandtabs()` existed. Changed that, then added and modified the tests.

**Pull request merged**